lib/billy/handlers/request_handler.rb

```diff
@@ -14,31 +15,42 @@ def handlers
     end
 
     def handle_request(method, url, headers, body)
+      request = request_log.record(method, url, headers, body)
+
       # Process the handlers by order of importance
       [:stubs, :cache, :proxy].each do |key|
         if (response = handlers[key].handle_request(method, url, headers, body))
+          @request_log.complete(request, key)
           return response
         end
       end
 
       body_msg = Billy.config.cache_request_body_methods.include?(method) ? " with body '#{body}'" : ''
+      request_log.complete(request, :error)
```

Comment:
I'm currently using this information to spit out a nice table via TablePrint:

```ruby
    TablePrint::Printer.table_print(requests, [
      :status,
      :handler,
      :method,
      { url: { width: 150 } },
      :headers,
      :body
    ])
```

This gives me a nice table such as:

```
STATUS   | HANDLER | METHOD  | URL                            | HEADERS                        | BODY
---------|---------|---------|--------------------------------|--------------------------------|-----
complete | proxy   | GET     | http://127.0.0.1:9887/example  | {"Accept"=>"text/html,appli... |
complete | proxy   | GET     | http://127.0.0.1:9887/example  | {"Accept"=>"text/css,*/*;q=... |
complete | proxy   | GET     | http://127.0.0.1:9887/example  | {"Referer"=>"http://127.0.0... |
complete | proxy   | GET     | http://127.0.0.1:9887/example  | {"User-Agent"=>"Mozilla/5.0... |
complete | proxy   | GET     | http://127.0.0.1:9887/example  | {"Referer"=>"http://127.0.0... |
inflight |         | GET     | http://127.0.0.1:9887/example  | {"Referer"=>"http://127.0.0... |
inflight |         | GET     | http://127.0.0.1:9887/example  | {"Referer"=>"http://127.0.0... |
inflight |         | GET     | http://127.0.0.1:9887/example  | {"Referer"=>"http://127.0.0... |
inflight |         | GET     | http://127.0.0.1:9887/example  | {"Referer"=>"http://127.0.0... |
inflight |         | GET     | http://127.0.0.1:9887/example  | {"Referer"=>"http://127.0.0... |
complete | error   | OPTIONS | https://example.com:443/exa... | {"Access-Control-Request-Me... |
```

I can immediately read `:error` from this output, but perhaps it's not semantically accurate. Should it maybe be `:unhandled` instead ?

Any preferences here @ronwsmith 🤔

Reply:
I'm okay with `:error` since we're using `error` for the below logging line already.